src/render/texture_array_cache.rs

```diff
@@ -206,12 +204,15 @@ impl TextureArrayCache {
                         });
 
                     for i in 0..count {
-                        let columns = (texture_size.x as f32 + spacing.x as f32)
-                            / (tile_size.x as f32 + spacing.x as f32);
-                        let sprite_sheet_x: f32 =
-                            (i as f32 % columns).floor() * (tile_size.x + spacing.x) as f32;
-                        let sprite_sheet_y: f32 =
-                            (i as f32 / columns).floor() * (tile_size.y + spacing.y) as f32;
+                        let columns = ((texture_size.x as f32)
+                            / (tile_size.x as f32 + spacing.x as f32))
+                            .floor();
+                        let sprite_sheet_x: f32 = (i as f32 % columns).floor()
+                            * (tile_size.x + spacing.x) as f32
+                            + spacing.x;
+                        let sprite_sheet_y: f32 = (i as f32 / columns).floor()
+                            * (tile_size.y + spacing.y) as f32
+                            + spacing.y;
```

Comment:
Don't divide by 2 here. This is because spacing is treated as the "space" between tiles. This is what tiled map editor does and I think it's something we should stick with.